apimonitor/views.py

```diff
@@ -30,6 +31,76 @@ def get_queryset(self):
+            'url': request.data.get('url'),
+            'schedule': request.data.get('schedule'),
+            'body_type': request.data.get('body_type')
+        }
+        api_monitor_serializer = APIMonitorSerializer(data=monitor_data)
+        monitor_obj = APIMonitor.objects.get(pk=kwargs['pk'])
+        if (api_monitor_serializer.is_valid()):
+            # Saved
+            monitor_obj.name = monitor_data['name']
+            monitor_obj.method = monitor_data['method']
+            monitor_obj.url = monitor_data['url']
+            monitor_obj.schedule = monitor_data['schedule']
+            monitor_obj.body_type = monitor_data['body_type']
+            monitor_obj.save()
+
+            # Delete old objects
+            APIMonitorQueryParam.objects.filter(monitor=kwargs['pk']).delete()
+            APIMonitorHeader.objects.filter(monitor=kwargs['pk']).delete()
+            APIMonitorBodyForm.objects.filter(monitor=kwargs['pk']).delete()
+            APIMonitorRawBody.objects.filter(monitor=kwargs['pk']).delete()
+
+            # Create new query param
+            for i in range(len(request.data.get('query_params', []))):
+                # Empty key or value will never reach backend, thanks Hugo
+                key = request.data.get('query_params')[i]['key']
+                value = request.data.get('query_params')[i]['value']
+                record = {
+                    "monitor": monitor_obj,
+                    "key": key,
+                    "value": value
+                }
+                APIMonitorQueryParam.objects.create(**record)
+
+            for i in range(len(request.data.get('headers', []))):
+                key = request.data.get('headers')[i]['key']
+                value = request.data.get('headers')[i]['value']
+                record = {
+                    "monitor": monitor_obj,
+                    "key": key,
+                    "value": value
+                }
+                APIMonitorHeader.objects.create(**record)
+
+            if (monitor_data['body_type'] == "FORM"):
+                for i in range(len(request.data.get('body_form', []))):
+                    key = request.data.get('body_form')[i]['key']
+                    value = request.data.get('body_form')[i]['value']
+                    record = {
+                        "monitor": monitor_obj,
+                        "key": key,
+                        "value": value
+                    }
+                    APIMonitorBodyForm.objects.create(**record)
+            elif (monitor_data['body_type'] == "RAW"):
+                record = {
+                    'monitor': monitor_obj,
+                    'body': request.data['raw_body']
+                }
+                APIMonitorRawBody.objects.create(**record)
+            serializer = APIMonitorSerializer(monitor_obj)
+            return Response(serializer.data)
+        else:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
sepertinya bisa ditambahin message errornya juga luck biar konsisten sama yg create
```suggestion
            return Response(data={"error": "['Please make sure your [name, method, url, schedule, body_type] is valid']"}, status=status.HTTP_400_BAD_REQUEST)
```

Reply:
Ini nanti aku urus di frontend Hugo, soalnya kemungkinan gagalnya cuma 1. Tapi kalau pada setuju, aku ganti sih, apakah menurut kalian penting? @andrew-susanto @luthfifahlevi @ferdifdi

Reply:
Setelah reconsider ulang, ini aku urus di backend aja sesuai idemu hugo wkwkwk, :) makasih